docs/articles/persistence/custom-persistence-provider.md

```diff
@@ -0,0 +1,467 @@
+    # as many messages as it has accumulated since the last write.
+    max-message-batch-size = 200
+
+    # If there is more time in between individual events gotten from the Journal
+    # recovery than this the recovery will fail.
+    # Note that it also affect reading the snapshot before replaying events on
+    # top of it, even though iti is configured for the journal.
+    recovery-event-timeout = 30s
+
+    circuit-breaker {
+        max-failures = 10
+        call-timeout = 10s
+        reset-timeout = 30s
+    }
+
+    # The replay filter can detect a corrupt event stream by inspecting
+    # sequence numbers and writerUuid when replaying events.
+    replay-filter {
+        # What the filter should do when detecting invalid events.
+        # Supported values:
+        # `repair-by-discard-old` : discard events from old writers,
+        #                           warning is logged
+        # `fail` : fail the replay, error is logged
+        # `warn` : log warning but emit events untouche
+        # `off` : disable this feature completely
+        mode = repair-by-discard-old
+
+        # It uses a look ahead buffer for analyzing the events.
+        # This defines the size (in number of events) of the buffer.
+        window-size = 100
+
+        # How many old writerUuid to remember
+        max-old-writers = 10
+
+        # Set this to `on` to enable detailed debug logging of each
+        # replayed event.
+        debug = off
+    }
+}
+```
+
+#### Pre-Start Requirement
+
+It is a good practice to allow user of your custom plugin to be able to auto initialize their back end event source from a blank slate. In order to do that, it is good practice to support `auto-initialize` setting in your journal HOCON settings.
+
+In order to support this, we will have to be able to stash all incoming messages while we're creating the tables in the event source. Here's an implementation that we use in our example code:  
+
+[!code-csharp[Startup](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=Startup "Journal startup sequence")]
+
+## Implementing Akka.Persistence SnapshotStore
+
+### Required Method Implementations
+
+#### LoadAsync
+
+```c#
+Task<SelectedSnapshot> LoadAsync(string persistenceId, SnapshotSelectionCriteria criteria)
+```
+
+* **`persistenceId`**: Identification of the persistent actor
+* **`criteria`**: Selection criteria for event loading
+
+This call is protected with a circuit-breaker.
+
+Asynchronously loads a snapshot.
+
+##### Code Sample
+
+`SelectSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[SelectSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=SelectSnapshotSql "SelectSnapshotSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[LoadAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=LoadAsync "LoadAsync method implementation")]
+
+#### SaveAsync
+
+```c#
```

Comment:
Do we leave this as it is or make use of the range selection though the line number may change in the future?

Reply:
see above